libkernelflinger/android.c

```diff
@@ -1169,9 +1295,11 @@ static EFI_STATUS setup_command_line(
         if (EFI_ERROR(ret))
                 goto out;
 
-        ret = add_bootvars(bootimage, &cmdline16);
-        if (EFI_ERROR(ret))
-                goto out;
+        if (aosp_header->header_version < BOOT_HEADER_V3) {
```

Comment:
any reason to add this check?

Reply:
bootvars are extracted from the second stage of boot image, this concept only exists in the header version below v3. It's not compatible with the header version v3 and v4.